drivers/network/dd/e1000/hardware.c

```diff
@@ -189,32 +190,74 @@ static BOOLEAN E1000ReadMdic(IN PE1000_ADAPTER Adapter, IN ULONG Address, USHORT
 }
 #endif
 
-
 static BOOLEAN E1000ReadEeprom(IN PE1000_ADAPTER Adapter, IN UCHAR Address, USHORT *Result)
 {
     ULONG Value;
     UINT n;
 
-    E1000WriteUlong(Adapter, E1000_REG_EERD, E1000_EERD_START | ((UINT)Address << E1000_EERD_ADDR_SHIFT));
+    if (Adapter->E1000_EERD_ADDR_SHIFT == 0xffff) {
+        NDIS_DbgPrint(MIN_TRACE, ("E1000DetermineAndSaveEERDLayout not run or wasn't successfull!\n"));
+        return FALSE;
+    }
+    E1000WriteUlong(Adapter, E1000_REG_EERD, E1000_EERD_START | ((UINT)Address << Adapter->E1000_EERD_ADDR_SHIFT));
 
     for (n = 0; n < MAX_EEPROM_READ_ATTEMPTS; ++n)
     {
         NdisStallExecution(5);
 
         E1000ReadUlong(Adapter, E1000_REG_EERD, &Value);
 
-        if (Value & E1000_EERD_DONE)
+        if (Value & Adapter->E1000_EERD_DONE_BIT)
             break;
     }
-    if (!(Value & E1000_EERD_DONE))
+    if (!(Value & Adapter->E1000_EERD_DONE_BIT))
     {
-        NDIS_DbgPrint(MIN_TRACE, ("EEPROM Read incomplete\n"));
+        NDIS_DbgPrint(MIN_TRACE, ("EEPROM Read incomplete! %x \n", Value));
         return FALSE;
     }
     *Result = (USHORT)(Value >> E1000_EERD_DATA_SHIFT);
+    NDIS_DbgPrint(MIN_TRACE, ("Read: %02x %04x\n", Address, *Result));
     return TRUE;
 }
 
+/* 8254x and 8257x have different layouts of EERF
+   Get it by trying to read the VID-field.
+*/
+static BOOLEAN E1000DetermineAndSaveEERDLayout(IN PE1000_ADAPTER Adapter) {
+    BOOLEAN success;
+    USHORT Result = 0;
+
+    //First, try the 8254x layout
```

Comment:
From a casual glance, the preferred comment style here seems to be `/* This */`, please adjust (or correct me if I'm wrong :p)

Reply:
Changed.